next.config.js

```diff
@@ -5,7 +5,31 @@ const nextConfig = {
   pageExtensions: ['page.tsx'],
   images: {
     unoptimized: true,
-  }
+  },
+  webpack(config) {
```

Comment:
Why is this change required?

Reply:
This change is required to optimize how we manage and render icons across our website. Currently, we're fetching icons as images from the server at runtime, which leads to several inefficiencies:

1. **Unnecessary Network Requests**: Every icon is fetched as a separate image file, increasing the number of network requests. This impacts load times and server performance.
2. **Complex Theme Management**: To handle different themes, we're using multiple useEffect hooks to fetch and update icons based on the current theme. This approach is repetitive and adds unnecessary complexity to our code.

By switching to SVGR, we can embed SVGs directly as React components. This allows us to:

1. **Reduce Network Requests**: Icons are embedded at build time, eliminating the need to fetch them from the server during runtime.
2. **Simplify Theme Management**: We can easily manipulate icon styles natively with CSS, like using `dark:fill-black` to change the fill color, instead of relying on multiple useEffect hooks.

Reply:
Such as #850, currently updating SVGs with theme would require `useTheme` and `useEffect` hooks, which again is repetitive and leads to inconsistencies sometimes. This would all be as simple as adding `dark:` classes to the svg components if we use svg's as react components and embed them during build time.

Reply:
Is this _necessary_ for this PR to work? If not, it should be in a separate change.

The effort is appreciated, but if it's unrelated it needs to be addressed as such.

Reply:
Kind of but I can change it! The theming with `useEffect` was a headache so I tried this as a fix. The icons in the PR depends on this but they are few so I can easily switch back. Lemme know!

Reply:
I'm okay if it's related.  I'll let @benjagm make the call.

Reply:
Sure! Just to be clear, this does not affect any existing svg bundling with webpack because all current svgs use public path of the deployed page to point to the resource and instead of javascript modules `import`.

Reply:
Thanks everyone for the great discussions. Let's continue with the change here.